### ISSUE TITLE
Inconsistent behaviour with Node text contents

### ISSUE THREAD

**Issue by Jameskmonger** [airbnb/enzyme#812]
I have noticed that there is a bit of an inconsistency with the way that text contents are handled.

I have a `span` tag which will be like so:

- `<span>50%</span>`
- `<span>70%</span>`
- `<span>25%</span>`

## Working behaviour

It works correctly when they are passed in as a single variable:

```javascript
wrapper.first().contains(<span>{ `${percentage}%` }</span>) // true

wrapper.first().contains(React.createElement("span", null, percentage + "%")) // transpiled
```

## Odd behaviour

When they are passed in as separate variables,

```javascript
wrapper.first().contains(<span>{ percentage }%</span>) // false

wrapper.first().contains(React.createElement("span", null, percentage, "%")) // transpiled
```

## Reasons

I can see that when they are passed in separately, the `React.createElement` call has two children. However, I feel that the expected behaviour of Enzyme would be to recognise this and return `true` in both cases.

**Comment by ljharb** [comment]
Is this with `mount` or `shallow`?

**Comment by Jameskmonger** [comment]
This is with `shallow`.

**Comment by nfcampos** [comment]
fixed by #512, released on 2.8.0
